test/spec/controllers/exportimportModal.js

```diff
@@ -0,0 +1,191 @@
+        {'model': 'Users','record': { '_id':'1569fe9d8b6addd83l559fd3', 'firstname': 'Ordinary', 'surname': 'User', 'email': 'normal@openim.org', 'passwordAlgorithm': 'sample/api', 'passwordHash': '539aa778930879b01b37ff62', 'passwordSalt': '79b01b37ff62', 'groups': ['limited'] },'status': 'Valid','message': '', 'uid': 'normal@openim.org'},
+        {'model': 'Mediators','record': { '_id':'4444fe9d8b6addd83l5595555', 'urn': 'AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE', 'version': '0.0.1','name': 'Test 1 Mediator','description': 'Test 1 Description','defaultChannelConfig': [{ 'name': 'Mediator Channel 1', 'urlPattern': '/channel1', 'routes': [{ 'name': 'Route 1', 'host': 'localhost', 'port': '1111', 'primary': true, 'type': 'http' }], 'allow': [ 'xdlab' ], 'type': 'http' }],'endpoints': [{ 'name': 'Route 1', 'host': 'localhost', 'port': '1111', 'primary': true, 'type': 'http' }]},'status': 'Conflict','message': '', 'uid': 'AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE'}
+      ]
+    };
+
+    createController = function() {
+      return $controller('ExportImportModalCtrl', { 
+        $scope: scope,
+        $modalInstance: modalInstance,
+        data: data
+      });
+    };
+
+  }));
+
+  afterEach(function() {
+    httpBackend.verifyNoOutstandingExpectation();
+    httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should attach a list of conflicts to the scope', function () {
+    createController();
+
+    scope.conflicts.length.should.equal(4);
+    scope.validImports.length.should.equal(2);
+
+  });
+
+  it('should execute saveImport() and import the data', sinon.test(function () {
+    createController();
+
+    // used to check if 
+    var spy = this.spy(modalInstance, 'close');
+
+    // lets change some data in our data object before doing the import
+    scope.conflicts.forEach(function(item){
+      item.action = 'overwrite';
+    });
+
+    // execute the import function
+    scope.saveImport(function() {
+      sinon.assert.calledOnce(spy);
+    });
+
+    httpBackend.flush();
+  }));
+
+  it('should execute saveImport() and do not import data', sinon.test(function () {
+    data.rows.forEach(function(item){
+      item.status = 'Conflict';
+    });
+
+    createController();
+    scope.conflicts.length.should.equal(6);
+
+    // used to monitor when modal gets closed - indicates completion
+    var spy = this.spy(modalInstance, 'close');
+
+    // lets change some data in our data object before doing the import
+    scope.conflicts.forEach(function(item){
+      item.action = 'ignore';
+    });
+
+    // execute the import function
+    scope.saveImport(function() {
+      sinon.assert.notCalled(spy);  
```

Comment:
It seems to me like 'close' should still be called if everything is ignored. Else you get a blank screen.